### DISCUSSION TITLE
How to call correctly callback_query

### DISCUSSION THREAD

**Discussion by Alheimurine** [telegraf/telegraf#1910] [Q&A]
Have 2 buttons. 
If user press 1st button - scene steps 1-2-3-4 (default scene steps).
But if user press 2nd button -  selected step 5 (and so on scene), while skipped 1 - 2 - 3. 
Need at press 2nd button, select step 5 and call `callback_query`. 
In my code correct skip to  step 5, but `callback_query` call only after double click

Handling button with skip steps 
```
if (ctx.update.callback_query.data === 'stock') {
            return ctx.wizard.selectStep(5)
        } else { 
        other code
}
```
and step 5 on this wizard.scene
```
stoName.on(['message', 'callback_query'], async (ctx) => {
    try {
        if (ctx.message) {
            console.log('scene 5 text')
            ctx.session.stoInfo.problem = ctx.message.text
            await ctx.reply('name.')
        } else if (ctx.callbackQuery) {
            console.log('scene 5 btn')
            ctx.session.stoInfo.reqType = 'isStock'
            await ctx.reply('name.')
        }
        return ctx.wizard.next()
    } catch(e) {
        console.error(e)
    }
})
```
Friends, help me please.

**Comment by 0xHal** [comment]
What's the problem here please?
Could you explain what are you trying to do?

**Comment by MKRhere** [comment]
If you want the selected step to be run right away, you should call it.

```TS
(ctx, next) => {
	...
	if (ctx.callbackQuery?.data === "stock") {
		ctx.wizard.selectStep(5);
		return ctx.wizard.step(ctx, next);
	}
```
